base/broadcast.jl

```diff
@@ -342,6 +342,47 @@ cat_nested(t::Broadcasted, rest...) = (cat_nested(t.args...)..., cat_nested(rest
 cat_nested(t::Any, rest...) = (t, cat_nested(rest...)...)
 cat_nested() = ()
 
+"""
+    RecursiveInliningEnforcerA
+
+A helper type for the `make_makeargs` function that is used to break
+method recursion that sometimes prevents inlining which results in increased
+compilation and execution times. See issue
+https://github.com/JuliaLang/julia/issues/41090 for details.
+"""
+struct RecursiveInliningEnforcerA{T}
+    makeargs::T
+end
+
+"""
+    RecursiveInliningEnforcerB
+
+A helper type for the `make_makeargs` function. See `RecursiveInliningEnforcerA`
+for details.
+"""
+struct RecursiveInliningEnforcerB{TMT,TMH,TT,TH,TF}
+    makeargs_tail::TMT
+    makeargs_head::TMH
+    headargs::TH
+    tailargs::TT
+    f::TF
+end
+
+for UB in [Any, RecursiveInliningEnforcerA]
+    @eval @inline function (bb::RecursiveInliningEnforcerB{TMT,TMH,TT,TH,TF})(args::Vararg{Any,N}) where {N,TMT,TMH<:$UB,TT,TH,TF}
```

Comment:
Why does this need two methods? Doesn't the `Any` bound cover everything?

Reply:
IIUC the compiler currently has problems with inlining recursive method calls, so by having two different methods doing the same thing there is less recursion to handle. There is no other reason to having two methods there. I don't know how exactly this works so I may be wrong though -- but it works.